From 66ea5f8b43eb92aa7723487dba10de79b02f0b0d Mon Sep 17 00:00:00 2001
From: Xiaoliang Yang <xiaoliang.yang_1@nxp.com>
Date: Fri, 23 Aug 2019 14:05:54 +0800
Subject: [PATCH 658/741] felix: TSN: fix stream identify set issue

commit 1822267f82a28075490c478722d430e9561531e5 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

When there has an exist dest mac and VID in mac table which is learned
by switch, stream identify set could not set the SFID for qci sfi module
and SSID for redundancy module. This patch added the SFID and SSID set
in stream identify set.

Signed-off-by: Xiaoliang Yang <xiaoliang.yang_1@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/ethernet/mscc/felix_tsn.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/net/ethernet/mscc/felix_tsn.c b/drivers/net/ethernet/mscc/felix_tsn.c
index 6041c2c26ac2..d733e699ddf4 100644
--- a/drivers/net/ethernet/mscc/felix_tsn.c
+++ b/drivers/net/ethernet/mscc/felix_tsn.c
@@ -756,6 +756,13 @@ int felix_cb_streamid_set(struct net_device *ndev, u32 index, bool enable,
 		return 0;
 	}
 
+	ocelot_write(ocelot,
+		     ANA_TABLES_STREAMDATA_SFID_VALID |
+		     ANA_TABLES_STREAMDATA_SFID(sfid) |
+		     ANA_TABLES_STREAMDATA_SSID_VALID |
+		     ANA_TABLES_STREAMDATA_SSID(ssid),
+		     ANA_TABLES_STREAMDATA);
+
 	reg = ocelot_read(ocelot, ANA_TABLES_MACACCESS);
 	dst_idx = ANA_TABLES_MACACCESS_DEST_IDX_X(reg);
 	ocelot_write(ocelot, ANA_TABLES_MACACCESS_VALID |
-- 
2.17.1

